### PR TITLE
Feature/resize service

### PR DESCRIPTION
**Pull request by DannyvanHolten**

### CODE REVIEW

File: resources/ResizeService.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    }
+
+    /**
+     * Get the attachment's ID by URL
+     *
+     * @param $file
+     * @return mixed
+     */
+    private function getAttachtmentIdByUrl($file)
```
Comment (HarwinBorger) [inline resources/ResizeService.php]: `getAttachtmentIdByUrl` should be `getAttachtmenIdByUrl`

File: resources/ResizeService.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * Catch the intermediate_image_sizes_advanced filter.
+     * And make sure all the resizing goes trough the ImageFocus crop service.
+     *
+     * @param $sizes
+     * @param $metadata
+     */
+    public function resizeAttachments($sizes, $metadata)
+    {
+        // The filter doesn't pass trough the attachment's ID. So we need to get it by url
```
Comment (HarwinBorger) [inline resources/ResizeService.php:35]: That sounds very error-sensitive, is there no other way?

File: resources/ResizeService.php (hunk 1 above)
Comment (HarwinBorger) [inline resources/ResizeService.php]: `getAttachtmentIdByUrl` should be `getAttachmentIdByUrl`

File: resources/ResizeService.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     * @return mixed
+     */
+    private function getAttachtmentIdByUrl($file)
+    {
+        global $wpdb;
+        $attachment = $wpdb->get_col(
+            $wpdb->prepare("SELECT ID FROM $wpdb->posts WHERE guid='%s';", wp_upload_dir()['baseurl'] . '/' . $file)
+        );
+
+        return $attachment[0];
```
Comment (HarwinBorger) [inline resources/ResizeService.php]: what happens if there is no result? $attachment[0] probably gives an error like: error offset blabla

File: resources/CropService.php [hunk 4]
```diff
@@ -98,7 +104,11 @@ private function getAttachment($attachmentId)
      */
     private function setFocusPoint($focusPoint)
     {
-        $this->focusPoint = $focusPoint;
+        $this->focusPoint = [50, 50];
```
Comment (HarwinBorger) [inline resources/CropService.php]: Please write down this variable `[50, 50]` as a private variable outside the function and in the top of the class. Should be shared with JS later on as well

File: resources/ResizeService.php (hunk 3 above)
Comment (DannyvanHolten) [reply resources/ResizeService.php]: it is impossible for this method not to have a [0] element. But I will improve it.

File: resources/ResizeService.php (hunk 2 above)
Comment (DannyvanHolten) [reply resources/ResizeService.php:35]: It is the only way.

**Review by HarwinBorger — APPROVED** [review]

**Pull request merged**